Documentation: Improvements to SAML configuration steps (#67437)
grafana/grafana: docs/sources/setup-grafana/configure-security/configure-authentication/saml/index.md

The SAML SSO standard uses asymmetric encryption to exchange information between the SP (Grafana) and the IdP. To perform such encryption, you need a public part and a private part. In this case, the X.509 certificate provides the public part, while the private key provides the private part. The private key needs to be issued in a [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.

> > **Note:** Directly supplying a certificate and private key to Grafana is optional. Commonly, the certificate and key are embedded in the IDP metadata (see that configuration step below), and refreshed as needed by Grafana automatically. This can save you the work of manually rotating your certs and keys whenever they expire.

Reviewer: ```suggestion
> **Note:** Directly supplying a certificate and private key to Grafana is optional. Commonly, the certificate and key are embedded in the IDP metadata (see the configuration step below) and refreshed as needed by Grafana automatically. This can save you the work of manually rotating your certs and keys whenever they expire.
```

> Whether supplying certificate and key directly or via idp_metadata: loading certs into Grafana helps ensure that Grafana is connecting to a verified endpoint (for example in case of MITM takeover scenarios).

Reviewer: ```suggestion
Whether supplying certificate and key directly or via idp_metadata: loading certs into Grafana helps ensure that Grafana connects to a verified endpoint (for example, in case of MITM takeover scenarios).
```

If you are directly supplying the certificate and key, also note that Grafana supports two ways of specifying both the `certificate` and `private_key`:

- Without a suffix (`certificate` or `private_key`), the configuration assumes you've supplied the base64-encoded file contents.
- With the `_path` suffix (`certificate_path` or `private_key_path`), then Grafana treats the value entered as a file path and attempts to read the file from the file system.

> **Note:** You can only use one form of each configuration option. Using multiple forms, such as both `certificate` and `certificate_path`, results in an error.

> You should always work with your company's security team on setting up certificates and private keys. If you need to generate them yourself (i.e. in the short term, for testing purposes, etc.), we give you the example below of generating your own certificate and private key, including the step of ensuring that the key is generated with the [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.

Reviewer: ```suggestion
You should always work with your company's security team on setting up certificates and private keys. If you need to generate them yourself (such as in the short term, for testing purposes, etc.), follow the example below to generate your certificate and private key, including the step of ensuring that the key is generated with the [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.
```

$ base64 -w0 cert.pem > cert.pem.base64
```

> Convert the private key to [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format

Reviewer: ```suggestion
Run the following command to convert the private key to [PKCS#8](https://en.wikipedia.org/wiki/PKCS_8) format.
```
